source/grid/tria.cc

```diff
@@ -3166,8 +3166,11 @@ namespace internal
                             ExcInconsistentLineInfoOfLine(line_vertices.first,
                                                           line_vertices.second,
                                                           "boundary ids"));
-
-                line->set_boundary_id_internal(subcell_line.boundary_id);
+                // do not use the boundary id provided in subcell_line
+                // if it is an internal one
+                if (subcell_line.boundary_id !=
+                    numbers::internal_face_boundary_id)
+                  line->set_boundary_id_internal(subcell_line.boundary_id);
```

Comment:
Let's use `TriaAccessor::set_boundary_id` then.
```suggestion
                  line->set_boundary_id(subcell_line.boundary_id);
```

Reply:
Definitely makes sense.